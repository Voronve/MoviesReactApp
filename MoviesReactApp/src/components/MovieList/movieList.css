.moviesList {
    margin-top: 60px;
}