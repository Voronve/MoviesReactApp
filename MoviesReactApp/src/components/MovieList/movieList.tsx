import './movieList.css'
import { MovieTile } from '../MovieTile/movieTile';
import { genres } from '../MovieListStateHandler/movieListStateHandler';

interface movieListProps {
    /** Movie list*/
    list: {
        image: string;
        title: string;
        releaseYear: number;
        genres: string[];
        rating: number;
        duration: string;
        description: string;
    }[],
    /** callback function for movie tile been clicked*/
    movieClick: (title: string) => void;
}

/** Movie list section*/
export function MovieList({list, movieClick}: movieListProps) {

    return(
        <div className="moviesList">
            {
                list.map( movie => { 
                    return(
                        <MovieTile movieData={{
                            image: movie.image,
                            title: movie.title,
                            releaseYear: movie.releaseYear,
                            genres: movie.genres as genres[],
                            onClick: movieClick}}
                            key={movie.title}/>
                        )})
            }
        </div>
    );
}