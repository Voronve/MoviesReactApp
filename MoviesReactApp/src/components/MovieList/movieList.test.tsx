import { MovieList } from './movieList';
import { fireEvent, render, screen } from '@testing-library/react';

describe("Testing MovieList component", () => {
    const onClick = jest.fn();
    const movieListMock = [{
        "image": "Comedy/Barbie.jpg",
        "title": "Barbie",
        "releaseYear": 2023,
        "genres": ["COMEDY"],
        "rating": 8.9,
        "duration": "2h 34min",
        "description": "At vero eos et accusamus et iusto odio dignissimos ducimus qui blanditiis praesentium voluptatum deleniti atque corrupti quos dolores et quas molestias excepturi sint occaecati cupiditate non provident, similique sunt in culpa qui officia deserunt mollitia animi, id est laborum et dolorum fuga. Et harum quidem rerum facilis est et expedita distinctio. Nam libero tempore, cum soluta nobis est eligendi optio cumque nihil impedit quo minus id quod maxime placeat facere possimus, omnis voluptas assumenda est, omnis dolor repellendus. Temporibus autem quibusdam et aut officiis debitis aut rerum necessitatibus saepe eveniet ut et voluptates repudiandae sint et molestiae non recusandae. Itaque earum rerum hic tenetur a sapiente delectus, ut aut reiciendis voluptatibus maiores alias consequatur aut perferendis doloribus asperiores repellat."
    },
    {
        "image": "Comedy/GrandBudapest.jpg",
        "title": "GrandBudapest",
        "releaseYear": 2014,
        "genres": ["COMEDY"],
        "rating": 8.9,
        "duration": "2h 34min",
        "description": "At vero eos et accusamus et iusto odio dignissimos ducimus qui blanditiis praesentium voluptatum deleniti atque corrupti quos dolores et quas molestias excepturi sint occaecati cupiditate non provident, similique sunt in culpa qui officia deserunt mollitia animi, id est laborum et dolorum fuga. Et harum quidem rerum facilis est et expedita distinctio. Nam libero tempore, cum soluta nobis est eligendi optio cumque nihil impedit quo minus id quod maxime placeat facere possimus, omnis voluptas assumenda est, omnis dolor repellendus. Temporibus autem quibusdam et aut officiis debitis aut rerum necessitatibus saepe eveniet ut et voluptates repudiandae sint et molestiae non recusandae. Itaque earum rerum hic tenetur a sapiente delectus, ut aut reiciendis voluptatibus maiores alias consequatur aut perferendis doloribus asperiores repellat."
    }];

    test("Testing movie list rendering", () => {

        const { asFragment } = render(<MovieList list={movieListMock} movieClick={onClick} />);

        expect(asFragment()).toMatchSnapshot();
    });

    test("Testing that callback function is working as expected", () => {
        render(<MovieList list={movieListMock} movieClick={onClick} />);
        const chosenMovie = movieListMock[1];

        fireEvent.click(screen.getByText(chosenMovie.title));

        expect(onClick).toBeCalledWith(chosenMovie.title);
    });
});