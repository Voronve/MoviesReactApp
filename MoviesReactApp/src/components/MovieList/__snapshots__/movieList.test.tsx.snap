// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing MovieList component Testing movie list rendering 1`] = `
<DocumentFragment>
  <div
    class="moviesList"
  >
    <div
      class="movieTile"
    >
      <img
        alt="Barbie"
        src="/media/Comedy/Barbie.jpg"
        width="322px"
      />
      <div
        class="movieInfo"
      >
        <div
          class="flex-container"
        >
          <div
            class="name"
          >
            Barbie
          </div>
          <div
            class="year"
          >
            2023
          </div>
        </div>
        <div
          class="ganres"
        >
          COMEDY
        </div>
      </div>
    </div>
    <div
      class="movieTile"
    >
      <img
        alt="GrandBudapest"
        src="/media/Comedy/GrandBudapest.jpg"
        width="322px"
      />
      <div
        class="movieInfo"
      >
        <div
          class="flex-container"
        >
          <div
            class="name"
          >
            GrandBudapest
          </div>
          <div
            class="year"
          >
            2014
          </div>
        </div>
        <div
          class="ganres"
        >
          COMEDY
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
